lib/oli/delivery/sections.ex

```diff
@@ -273,21 +274,36 @@ defmodule Oli.Delivery.Sections do
         e in Enrollment,
         join: s in Section,
         on: e.section_id == s.id,
-        where: s.slug == ^section_slug and s.status != :deleted,
+        where: s.slug == ^section_slug and s.status == :active,
         preload: [:user, :context_roles],
         select: e
       )
 
     Repo.all(query)
   end
 
+  @doc """
+  Returns true if there is student data associated to the given section.
+  """
+  def has_student_data?(section_slug) do
+    query =
+      from(
+        snapshot in Snapshot,
+        join: s in assoc(snapshot, :section),
+        where: s.slug == ^section_slug and s.status == :active,
```

Comment:
It seems wrong to include "status == `:active`" within this query.   If `has_student_data?` was ever called on a section that was archived and that actually had student data, it would return `false`, and perhaps lead data being deleted.

Reply:
Actually there's no way for a user to "delete" an archived section in Torus now, but I see your point and this logic doesn't seem semantically right. I pushed the changes in the last commit, thanks!